scanpy/_utils/__init__.py

```diff
@@ -483,24 +486,56 @@ def update_params(
 # --------------------------------------------------------------------------------
 
 
-def check_nonnegative_integers(X: Union[np.ndarray, sparse.spmatrix]):
+_FlatSupportedArray = Union[np.ndarray, sparse.spmatrix, DaskArray]
+
+
+def get_ufuncs(data: np.ndarray | DaskArray):
+    if isinstance(data, DaskArray):
+        import dask.array as da
+
+        return da
+    return np
+
+
+def check_nonnegative_integers(X: _FlatSupportedArray) -> bool:
     """Checks values of X to ensure it is count data"""
     from numbers import Integral
 
-    data = X if isinstance(X, np.ndarray) else X.data
+    ufuncs = get_ufuncs(X)
+    data = _get_values(X)
```

Comment:
Until we have other cases, can this be a branch? I feel like it's a lot of indirection for this branch.

```suggestion
    # get (possibly nonzero) values as (possibly flat) array
    data = X if not sparse.issparse(x) else X.data
```